### PR TITLE
tools: add common makefile for mspdebug programmer

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR introduces a common makefile for defining the mspdebug programmer variables. Related boards (only msb430 and msb430h apparently) are adapted to make use of it.

Be careful, now the programmer is called `mspdebug` before it was `olimex` but it was a confusion with the mspdebug internal programmer used.

### Testing procedure

- Green CI
- msb430* boards are still flashable (I don't have the hardware to test this)

### Issues/PRs references

Will help in #15477

### CODE REVIEW

File: makefiles/tools/mspdebug.inc.mk [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+FLASHFILE ?= $(HEXFILE)
+FFLAGS = $(MSPDEBUGFLAGS) "prog $(FLASHFILE)"
+
+# setup debugger
+DEBUGSERVER = $(FLASHER)
+DEBUGSERVER_FLAGS = $(MSPDEBUGFLAGS) gdb
+DEBUGGER = $(PREFIX)gdb
+DEBUGGER_FLAGS = --ex="target remote localhost:2000" --ex "monitor reset halt" --ex load -ex "monitor reset halt"  $(ELFFILE)
+
+INCLUDES += -I$(RIOTBOARD)/common/msb-430/include
```
Comment (fjmolinas) [inline makefiles/tools/mspdebug.inc.mk:15]: How is this related?

**Review by fjmolinas — COMMENTED** [review]
Looks good other than the include that seems weirdly placed, all other variables but the PROGRAMMER are the same:

```
BOARD=msb-430 make -C examples/hello-world/ --no-print-directory info-debug-variable-PROGRAMMER info-debug-variable-MSPDEBUGFLAGS info-debug-variable-FLASHER info-debug-variable-FFLAGS info-debug-variable-DEBUGSERVER info-debug-variable-DEBUGSERVER_FLAGS info-debug-variable-DEBUGGER info-debug-variable-DEBUGGER_FLAGS info-debug-variable-RESET info-debug-variable-RESET_FLAGS > master.txt
BOARD=msb-430 make -C examples/hello-world/ --no-print-directory info-debug-variable-PROGRAMMER info-debug-variable-MSPDEBUGFLAGS info-debug-variable-FLASHER info-debug-variable-FFLAGS info-debug-variable-DEBUGSERVER info-debug-variable-DEBUGSERVER_FLAGS info-debug-variable-DEBUGGER info-debug-variable-DEBUGGER_FLAGS info-debug-variable-RESET info-debug-variable-RESET_FLAGS > pr.txt
diff master.txt pr.txt 
1c1
< olimex
---
> mspdebug

```

File: makefiles/tools/mspdebug.inc.mk (hunk 1 above)
Comment (aabadie) [reply makefiles/tools/mspdebug.inc.mk:15]: This is a mistake :)

Comment (aabadie) [reply makefiles/tools/mspdebug.inc.mk]: Should be fixed now!

**Review by fjmolinas — APPROVED** [review]
ACK, please squash.
